src/components/MapView/Analyser/index.tsx

```diff
@@ -86,17 +119,69 @@ function Analyser({ classes }: AnalyserProps) {
   const adminBoundariesExtent = useMemo(() => {
     if (!boundaryLayerData)
       // not loaded yet. Should be loaded in MapView
-      return null;
+      return undefined;
     return calculateExtentFromGeoJSON(boundaryLayerData.data) as Extent; // we get extents of admin boundaries to give to the api.
   }, [boundaryLayerData]);
 
+  const hazardLayerOptions = map(hazardLayers, hazardLayer => {
```

Comment:
Nice! I like what you did here. But we can be a *little* more efficient by taking into advantage that `baselineLayer` and `hazardLayer` are very similar in types, so we could possibly make a function that just takes `CommonLayerProps[]` and makes a `FormControlLabel` for us. Kinda like `createRadioButtons(layers: CommonLayerProps[])`. We are only using `id` and `title`, which are common to both layer types.

Reply:
I will do it.